### DISCUSSION TITLE
5.2.0~alpha1: Format.pp_set_geometry cannot be set to Int.max_int anymore

### DISCUSSION THREAD

**Discussion by kit-ty-kate** [ocaml/ocaml#12986] [General]
https://github.com/ocaml/ocaml/pull/12477 added a check for each geometry when using `Format.pp_set_geometry` so that the margin wouldn't go above `pp_infinity`. This has the intended consequence of making such code fail at runtime:
```
# Format.pp_set_geometry Format.std_formatter ~max_indent:2 ~margin:Int.max_int;;
Exception: Invalid_argument "Format.pp_set_geometry: margin >= pp_infinity".
```
I've noticed this code used in the wild in at least https://github.com/RedPRL/asai (other users may not have surfaced yet due to the runtime nature of the failure) and I'm wondering what would be the best course of action for users that have this sort of code.
Replace `Int.max_int` by `1_000_000_000` and add a TODO comment asking to replace it in the future by `Format.pp_infinity` when the project requires at least OCaml 5.2 ?

**Comment by favonia** [comment]
This turns out to be more annoying than expected. One has to write `~margin:(Format.pp_infinity-1)` because the margin cannot be equal to `pp_infinity` either.

**Comment by Octachron** [comment]
Using `1_000_000_000` might be a good enough option.

I was wondering if the `set_geometry` function should clip the value during the a transition period, but the specification of the function is that it never tries to second guess the user (contrarily to the safe function).
